File: arcade/intro/digitsProduct/digitsProduct.java
int digitsProduct(int product) {
    if (product == 0) return 10;
    if (product < 10) return product;
    int p = product;
    String result = "";
    for (int i = 9; i > 1; i--)
        if (p % i == 0) {    
            p /= i; 
            result = i + result;
            i = 10;
        }
    return p == 1 ? new Integer(result) : -1;
}

File: arcade/intro/digitsProduct/digitsProduct.py
def digitsProduct(product): 
    if product ==0:
        return 10
    if product < 10:
        return product
    p = product
    r = ''
    for i in range(9,1):
        if p % i == 0:
            p/= i
            r = i + r
            i = 9
            print(r)
        if p == 1: break
            
    if p == 1:
        return r
    else:
        return -1